test/Base_Tests/src/Semantic/Conversion_Spec.enso

```diff
@@ -543,6 +553,70 @@ add_specs suite_builder =
 
             accept ..Nothing . is_nothing . should_be_true
 
+        group_builder.specify "Autoscope_Type Raw" <|
+            r = ..Raw ..My_Other
+            x = r:Autoscope_Type
+            Meta.type_of x . should_equal Autoscope_Type
+            x.to_text . should_equal "(Raw ..My_Other)"
+
+        group_builder.specify "Autoscope_Type Typed" <|
+            r = ..Typed ..My_Other
+            x = r:Autoscope_Type
+            Meta.type_of x . should_equal Autoscope_Type
+            x.to_text . should_equal "(Typed ..My_Other)"
+
+        group_builder.specify "Autoscope_Type Complex1" <|
+            r = ..Complex1 ..My_Other
+            x = r:Autoscope_Type
+            Meta.type_of x . should_equal Autoscope_Type
+            x.to_text . should_equal "(Complex1 ..My_Other)"
+
+        group_builder.specify "Autoscope_Type Complex2" <|
+            r = ..Complex2 ..My_Other
+            x = r:Autoscope_Type
+            Meta.type_of x . should_equal Autoscope_Type
+            x.to_text . should_equal "(Complex2 ..My_Other)"
+
+        group_builder.specify "Autoscope_Type factory typed" <|
+            x = Autoscope_Type.f_typed ..My_Other
+            Meta.type_of x . should_equal Autoscope_Type
+            x.to_text . should_equal "(Raw ..My_Other)"
+
+        group_builder.specify "Autoscope_Type factory complex1" <|
+            x = Autoscope_Type.f_complex1 ..My_Other
+            Meta.type_of x . should_equal Autoscope_Type
+            x.to_text . should_equal "(Raw ..My_Other)"
+
+        group_builder.specify "Autoscope_Type factory complex2" <|
+            x = Autoscope_Type.f_complex2 ..My_Other
+            Meta.type_of x . should_equal Autoscope_Type
+            x.to_text . should_equal "(Raw ..My_Other)"
+
+
+        confused1 filter =
+            materialize f:Foo = f
+
+            case filter of
+                _:Function -> Panic.catch Any (materialize filter) err->
+                    [err, filter]
+                _ -> filter
+
+        group_builder.specify "Order of specializations 1 & 2" <|
+            fn x = x+2
+
+            r1 = confused1 fn
+            r2 = confused1 (..Value 4)
+            r = [r1, r2]
+            r.to_text . should_contain "Foo.Value 4"
+
+        group_builder.specify "Order of specializations 2 & 1" <|
```

Comment:
This test used to pass when executed in isolation:
```bash
sbt:enso> runEngineDistribution --run test/Base_Tests/src/Semantic/Conversion_Spec.enso Order.of.specializations.2
```
but when it was executed after `"Order of specializations 1 & 2"` then it would fail:
```
sbt:enso> runEngineDistribution --run test/Base_Tests/src/Semantic/Conversion_Spec.enso Order.of.specializations
```
81c01cbdea fixes both problems.